File: estudos-pf/nivel-1-map-filter-reduce/solution/q2/q2.py
def eh_negativo(numero: int) -> bool:
    return 1 if numero < 0 else 0 

def negativos(sequencia: list) -> list:
    return map(eh_negativo, sequencia)

def conta_negativos(sequencia: list) -> int:
    return sum(negativos(sequencia))

assert conta_negativos([]) == 0
assert conta_negativos([1, 2, 3]) == 0
assert conta_negativos([-1, -2, -3]) == 3
assert conta_negativos([0, -5]) == 1

File: estudos-pf/nivel-1-map-filter-reduce/solution/q2/q2.clj
(defn negativo? [numero]
  (if (< numero 0) 
        1
        0))

(defn conta_negativos [numeros]
  (reduce + (map negativo? numeros)))

(assert (= (conta_negativos []) 0))
(assert (= (conta_negativos [1 2 3]) 0))
(assert (= (conta_negativos [-1 -2 3]) 2))
(assert (= (conta_negativos [0 -5]) 1))
